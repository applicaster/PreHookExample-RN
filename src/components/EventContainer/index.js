import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  StyleSheet,
  View,
} from 'react-native';
import hexToRgb from 'hex-to-rgb';
import EventHeader from '../EventHeader';
import EventCaption from '../EventCaption';
import EventMedia from '../EventMedia';
import ActionBar from '../ActionBar';

const screenWidth = Dimensions.get('window').width;
const styles = StyleSheet.create({
  eventContainer: {
    borderBottomColor: '#696A6B',
    borderBottomWidth: StyleSheet.hairlineWidth,
    width: screenWidth,
  },
});

class EventContainer extends Component {
  getEventSeparatorStyles() {
    const rgb = hexToRgb(this.context.textColor || '#FFFFFF');
    const rgbaColor = `rgba(${rgb[0]},${rgb[1]},${rgb[2]}, 0.5)`;
    return { borderBottomColor: rgbaColor };
  }
  
  renderMedia() {
    const { type } = this.props.event;
    if (type === 'image' || type === 'video' || type === 'gallery') {
      const { url, height, width } = this.props.event.images.default;
      return <EventMedia imageUrl={url} width={width} height={height} />;
    }

    return null;
  }
  
  renderHeader() {
    const { source, type, user, createdAt } = this.props.event;
    const { avatarImageUrl, name, userName } = user;
    const overlayHeaderOnMedia = (type === 'image') || (type === 'video') || (type === 'gallery');

    return (<EventHeader
      avatarImageUrl={avatarImageUrl}
      createdAt={createdAt}
      name={name}
      overlay={overlayHeaderOnMedia}
      source={source}
      userName={userName}
    />);
  }

  renderCaption() {
    return <EventCaption caption={this.props.event.caption} />;
  }

  renderActionBar() {
    return (<ActionBar
      socialNetwork={this.props.event.source}
      likesCount={this.props.event.likesCount}
      commentsCount={this.props.event.commentsCount}
    />
    );
  }

  render() {
    const containerSeparatorColor = this.getEventSeparatorStyles();
    return (
      <View style={[styles.eventContainer, containerSeparatorColor]}>
        {this.renderHeader()}
        {this.renderMedia()}
        {this.renderCaption()}
        {this.renderActionBar()}
      </View>
    );
  }
}

EventContainer.propTypes = {
  event: PropTypes.object,
};

EventContainer.contextTypes = {
  textColor: PropTypes.string,
};

export default EventContainer;
