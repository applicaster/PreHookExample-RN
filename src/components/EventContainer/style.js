import { StyleSheet } from 'react-native';
import { BORDER_RADIUS } from '../../constants/measurements';

export const styles = StyleSheet.create({
  eventContainer: {
    borderRadius: BORDER_RADIUS, // TODO: MAKE THIS CONFIGURABLE
    overflow: 'hidden',
  },
});
