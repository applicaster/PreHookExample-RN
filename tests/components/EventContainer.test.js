import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventContainer from '../../src/components/EventContainer';
import { event } from './eventTestDataUtils';

describe('EventContainer', () => {
  test('component renders correctly with an image event', () => {
    const tree = renderer.create(
      <EventContainer event={event(1, 'image')} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with a text event', () => {
    const tree = renderer.create(
      <EventContainer event={event(1, 'text')} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

