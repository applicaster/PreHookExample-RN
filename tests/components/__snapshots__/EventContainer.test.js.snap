// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventContainer component renders correctly with a text event 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "marginBottom": 5,
        "marginHorizontal": 5,
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 12,
          "paddingTop": 12,
        },
        false,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "height": 43,
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "width": 220,
          },
          false,
        ]
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "600",
                "marginBottom": -2,
                "marginTop": -3,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          Test Name
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          @
          applicaster
        </Text>
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 12,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          August 4
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 9,
          "top": 9,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "feed_rn_instagram_icon",
          }
        }
        style={
          Array [
            Object {
              "height": 30,
              "width": 30,
            },
            Object {
              "tintColor": undefined,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 18,
          "marginBottom": 5,
          "marginHorizontal": 12,
          "marginTop": 17,
        },
        Object {
          "color": undefined,
        },
      ]
    }
  >
    Low. Tarmac T2 , CasCam, SPC LCAs.
Thanks 
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      @azpinstalls
    </Text>
     for the work, parts, advice and setup.

    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #brz
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #gt86
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #ft86
    </Text>
    
  </Text>
  <ActionBar
    commentsCount={0}
    eventId="someId"
    eventOriginUrl={undefined}
    likesCount={30}
    retweetCount={undefined}
    socialNetwork="instagram"
    textToShare="Low. Tarmac T2 , CasCam, SPC LCAs.
Thanks @azpinstalls for the work, parts, advice and setup.
#brz #gt86 #ft86"
  />
</View>
`;

exports[`EventContainer component renders correctly with an image event 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "marginBottom": 5,
        "marginHorizontal": 5,
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <EventMedia
    eventId="someId"
    height={504}
    imageUrl="https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20590092_108583273161709_7296810043293302784_n.jpg"
    videoUrl={undefined}
    width={640}
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 12,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "transparent",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 740,
          "zIndex": 2,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "height": 43,
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "width": 220,
          },
          false,
        ]
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "600",
                "marginBottom": -2,
                "marginTop": -3,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          Test Name
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          @
          applicaster
        </Text>
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 12,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          August 4
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 9,
          "top": 9,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "feed_rn_instagram_icon",
          }
        }
        style={
          Array [
            Object {
              "height": 30,
              "width": 30,
            },
            Object {
              "tintColor": undefined,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 18,
          "marginBottom": 5,
          "marginHorizontal": 12,
          "marginTop": 17,
        },
        Object {
          "color": undefined,
        },
      ]
    }
  >
    Low. Tarmac T2 , CasCam, SPC LCAs.
Thanks 
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      @azpinstalls
    </Text>
     for the work, parts, advice and setup.

    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #brz
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #gt86
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #ft86
    </Text>
    
  </Text>
  <ActionBar
    commentsCount={0}
    eventId="someId"
    eventOriginUrl={undefined}
    likesCount={30}
    retweetCount={undefined}
    socialNetwork="instagram"
    textToShare="Low. Tarmac T2 , CasCam, SPC LCAs.
Thanks @azpinstalls for the work, parts, advice and setup.
#brz #gt86 #ft86"
  />
</View>
`;
